source/_components/mcp23017.markdown

```diff
@@ -0,0 +1,121 @@
+
+The pin numbers are from 0 to 15 where: 0-7 correspond to port A (A1-A8) and 8-15 to port B (B1-B8).
+
+## {% linkable_title Configuration %}
+
+To use the I/O pins of an mcp23017 connected to and I2C bus of your Raspberry Pi as binary sensors, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+binary_sensor:
+  - platform: mcp23017
+    i2c_address: 0x20
+    pins:
+      0: PIR Office
+      1: PIR Bedroom
+```
+
+{% configuration %}
+i2c_address:
+  description: i2c address of MCP23017 chip.
+  required: false
+  type: integer
+  default: "`0x20`"
+pins:
+  description: List of used pins.
+  required: true
+  type: map
+  keys:
+    "pin: name":
+      description: The pin numbers (from 0 to 15) and corresponding names.
+      required: true
+      type: [integer, string]
+scan_interval:
+  description: Interval to scan for sensor state changes in seconds.
+  required: false
+  type: integer
+  default: 15
+invert_logic:
+  description: If `true`, inverts the output logic to ACTIVE LOW.
+  required: false
+  type: boolean
+  default: "`false` (ACTIVE HIGH)"
+pull_mode:
+  description: >
+    Type of internal pull resistor to use.
+    Options are `UP` - pull-up resistor and `DOWN` - pull-down resistor.
+  required: false
+  type: string
+  default: "`UP`"
+{% endconfiguration %}
+
+NOTE: MCP23017 only has internal pull-up resistors, if you want to use pull-down you will have to wire your own pull-down resistors.
+
+For more details about the MCP23017 I2C I/O port expander you can find its datasheet here: [MCP23017](https://www.microchip.com/wwwproducts/en/MCP23017).
+
+## {% linkable_title Switch %}
+
+The `mcp23017` switch platform allows you to write to the I/O pins of your [MCP23017 I2C I/O expander](https://www.adafruit.com/product/732).
+
+The pin numbers are from 0 to 15 where: 0-7 correspond to port A (A1-A8) and 8-15 to port B (B1-B8).
+
+## {% linkable_title Configuration %}
+
+To use the I/O pins of an mcp23017 connected to and I2C bus of your Raspberry Pi as switches, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+switch:
+  - platform: mcp23017
+    i2c_address: 0x20
+    pins:
+      11: Fan Office
+      12: Light Desk
+```
+
+{% configuration %}
+i2c_address:
+  description: i2c address of MCP23017 chip.
+  required: false
+  type: integer
+  default: "`0x20`"
+ports:
+  description: Array of used pins.
+  required: true
+  type: list
+  keys:
+    port:
+      description: The pin numbers (from 0 to 15) and corresponding names.
+      required: true
+      type: [integer, string]
+invert_logic:
+  description: If true, inverts the output logic to ACTIVE LOW.
+  required: false
+  default: false
+  type: boolean
+{% endconfiguration %}
+
+For more details about the MCP23017 I2C I/O port expander you can find its datasheet here: [MCP23017](https://www.microchip.com/wwwproducts/en/MCP23017).
```

Comment:
Maybe move this to the top so it is not repeated in each section.

Reply:
I made the suggested changes. I went through the documentation standards and there is nothing in regards to section levels. I used an existing page as a template but it does make more sense to have "Configuration" as a sub-section of "Binary Sensor/Switch" so I did that as well.